Microsoft.Toolkit.Uwp.UI.Controls/MarkdownTextBlock/MarkdownTextBlock.cs

```diff
@@ -21,32 +46,61 @@ public MarkdownTextBlock()
-            RegisterPropertyChangedCallback(SchemeListProperty, OnPropertyChanged);
+            FontSizePropertyToken = RegisterPropertyChangedCallback(FontSizeProperty, OnPropertyChanged);
+            BackgroundPropertyToken = RegisterPropertyChangedCallback(BackgroundProperty, OnPropertyChanged);
+            BorderBrushPropertyToken = RegisterPropertyChangedCallback(BorderBrushProperty, OnPropertyChanged);
+            BorderThicknessPropertyToken = RegisterPropertyChangedCallback(BorderThicknessProperty, OnPropertyChanged);
+            CharacterSpacingPropertyToken = RegisterPropertyChangedCallback(CharacterSpacingProperty, OnPropertyChanged);
+            FontFamilyPropertyToken = RegisterPropertyChangedCallback(FontFamilyProperty, OnPropertyChanged);
+            FontStretchPropertyToken = RegisterPropertyChangedCallback(FontStretchProperty, OnPropertyChanged);
+            FontStylePropertyToken = RegisterPropertyChangedCallback(FontStyleProperty, OnPropertyChanged);
+            FontWeightPropertyToken = RegisterPropertyChangedCallback(FontWeightProperty, OnPropertyChanged);
+            ForegroundPropertyToken = RegisterPropertyChangedCallback(ForegroundProperty, OnPropertyChanged);
+            PaddingPropertyToken = RegisterPropertyChangedCallback(PaddingProperty, OnPropertyChanged);
+            RequestedThemePropertyToken = RegisterPropertyChangedCallback(RequestedThemeProperty, OnPropertyChanged);
+
         }
 
-        private void ThemeListener_ThemeChanged(Helpers.ThemeListener sender)
+        private void OnUnloaded(object sender, RoutedEventArgs e)
         {
-            RenderMarkdown();
+            UnhookListeners();
+            themeListener.ThemeChanged -= ThemeListener_ThemeChanged;
+            themeListener = null;
+
+            // Register for property callbacks that are owned by our parent class.
+            UnregisterPropertyChangedCallback(FontSizeProperty, FontSizePropertyToken);
+            UnregisterPropertyChangedCallback(BackgroundProperty, BackgroundPropertyToken);
+            UnregisterPropertyChangedCallback(BorderBrushProperty, BorderBrushPropertyToken);
+            UnregisterPropertyChangedCallback(BorderThicknessProperty, BorderThicknessPropertyToken);
+            UnregisterPropertyChangedCallback(CharacterSpacingProperty, CharacterSpacingPropertyToken);
+            UnregisterPropertyChangedCallback(FontFamilyProperty, FontFamilyPropertyToken);
+            UnregisterPropertyChangedCallback(FontStretchProperty, FontStylePropertyToken);
+            UnregisterPropertyChangedCallback(FontWeightProperty, FontWeightPropertyToken);
+            UnregisterPropertyChangedCallback(ForegroundProperty, ForegroundPropertyToken);
+            UnregisterPropertyChangedCallback(PaddingProperty, PaddingPropertyToken);
+            UnregisterPropertyChangedCallback(RequestedThemeProperty, RequestedThemePropertyToken);
+            _rootElement = null;
```

Comment:
_rootElement probably should not be nulled here, because Loaded can be fired again after Unloaded without calling OnApplyTemplate in certain circumstances on the same instance of the control leading to a broken control as there's no reference to the _rootElement

(And to be fair, it does not need to nulled anyway as it has the same lifetime as MarkDownTextBlock and will get garbage collected with it / not cause a memory leak out of the box)